native/mod_manager/mod_manager.c

```diff
@@ -1229,12 +1264,97 @@ static char * process_config(request_rec *r, char **ptr, int *errtype)
         return MNODEET;
     }
 
+    /* Check for corresponding proxy_worker */
+    worker = proxy_node_getid(r, &nodeinfo, &id, &the_conf);
+    if (id != 0) {
+       /* Same node should be OK, different nodes will bring problems */
+       if (node != NULL && node->mess.id==id) {
+          ap_log_error(APLOG_MARK, APLOG_DEBUG, 0, r->server,
+                       "process_config: proxy_node_getid() worker exist and should be OK");
+       } else {
+          /* Here that is the tricky part, we will insert_update the whole node including proxy_worker_shared */
+          char *pptr;
+          unsigned long offset;
+          
+          ap_log_error(APLOG_MARK, APLOG_DEBUG, 0, r->server,
+                       "process_config: proxy_node_getid() worker %d (%s) exists and IS NOT OK!!!", id, nodeinfo.mess.JVMRoute);
+          if (node == NULL) {
+              /* try to read the node */
+              nodeinfo_t workernodeinfo;
+              nodeinfo_t *workernode;
+              workernodeinfo.mess.id = id;
+              workernode = read_node(nodestatsmem, &workernodeinfo);
+              if (workernode != NULL) {
+                  if (strcmp(workernode->mess.JVMRoute, "REMOVED")==0) {
+                      /* We are in the remove process */
+                      /* Something to clean ? */
+                      removed = -1;
+                      strcpy(workernode->mess.JVMRoute, nodeinfo.mess.JVMRoute);
+                  } else {
+                      /* if the workernode->mess is zeroed we are going to reinsert it */
+                      if (workernode->mess.JVMRoute[0] == '\0')
+                          removed = -1;
+                      else {
```

Comment:
I would prefer curly brackets around the `if`'s content even thought it's just one line. It seems dense to me and in general, personally I don't like mixing `if else` with and without brackets.